tensorflow_datasets/core/download/download_manager.py

```diff
@@ -421,6 +421,21 @@ def _find_existing_path(self, url: str, url_path: str) -> Optional[str]:
         self._recorded_url_infos[url] = _read_url_info(url_path)
     return existing_path
 
+  def _check_manually_downloaded(self, url):
+    """Checks if file is already downloaded in manual_dir."""
+    url_info = self._url_infos.get(url)
+    # If url info was recorded with filename
+    manually_downloaded_path = None
+    if url_info and url_info.filename:
+        manually_downloaded_path = os.path.join(
+          self._manual_dir,
+          url_info.filename
+        )
+        # File not manually downloaded
+        if not tf.io.gfile.exists(manually_downloaded_path):
```

Comment:
Could you also check that the checksums match `url_info.checksums` if `_force_checksums_validation` is True ?

Reply:
Hi, I have implemented the checksum check for the case when the user hasn't extracted the dataset. I'm working on the case when the user has extracted the dataset.

Thanks.